contracts/AssetPool.sol

```diff
@@ -253,13 +260,75 @@ contract AssetPool is Ownable {
         underwriterToken.burn(covAmount);
     }
 
+    /// @notice Withdraws collateral tokens to a new Asset Pool which previously
+    ///         was approved by the governance. Old underwriter tokens are burned
+    ///         in favor of new minted tokens that proof share of
+    ///         collateral tokens in a new Asset Pool.
+    function withdrawToNewAssetPool() external {
+        /* solhint-disable not-rely-on-time */
+        require(
+            address(newAssetPool) != address(0),
+            "New asset pool must be assigned"
+        );
+
+        uint256 covUnderwriterBalance = underwriterToken.balanceOf(msg.sender);
+
+        require(
+            covUnderwriterBalance > 0,
+            "Underwriter's token balance must be greater than 0"
+        );
+
+        uint256 covSupply = underwriterToken.totalSupply();
+
+        // slither-disable-next-line reentrancy-events
+        rewardsPool.withdraw();
+
+        uint256 collateralBalance = collateralToken.balanceOf(address(this));
+
+        uint256 amountToWithdraw =
+            covUnderwriterBalance.mul(collateralBalance).div(covSupply);
+
+        collateralToken.safeTransfer(address(newAssetPool), amountToWithdraw);
+        emit WithdrawalToNewAssetPoolCompleted(
+            msg.sender,
+            amountToWithdraw,
+            block.timestamp
+        );
+
+        // mappings cleanup since all the underwriter's funds are transferred to
+        // a new asset pool
+        delete withdrawalInitiatedTimestamp[msg.sender];
+        delete pendingWithdrawal[msg.sender];
+
+        underwriterToken.burn(covUnderwriterBalance);
+        newUnderwriterToken.mint(msg.sender, covUnderwriterBalance);
```

Comment:
I do not think we can assume COV amount minted by the second pool should be the same. It could happen that there was a claim and the math may not hold in this case.

Reply:
Do you mean that a claim could happen a transaction early? If yes, then a couple of lines above in this code we have the following, which reads the latest balance of the collateral token.
```
rewardsPool.withdraw(); 
uint256 collateralBalance = collateralToken.balanceOf(address(this));
```

Reply:
I do not think this helps. New asset pool and old asset pool have their own COV supplies and collateral token balances.